ocaml/typing/typecore.ml

```diff
@@ -5730,15 +5731,17 @@ and type_expect_
     pattern guard. See the comments on [match_info] for the meaning of the
     returned type.
 
+   [require_value_clause]: should [No_value_clauses] be raised if no value
```

Comment:
I think we should say "case" instead of "clause" to be consistent with the rest of the file.